x/oracle/keeper/owasm.go

```diff
@@ -152,3 +160,42 @@ func (k Keeper) ResolveRequest(ctx sdk.Context, reqID types.RequestID) {
 		k.ResolveSuccess(ctx, reqID, env.Retdata, output.GasUsed)
 	}
 }
+
+// CollectFee subtract fee from fee payer and send them to treasury
+func (k Keeper) CollectFee(ctx sdk.Context, payer sdk.AccAddress, feeLimit sdk.Coins, askCount uint64, rawRequests []types.RawRequest) (sdk.Coins, error) {
+
+	if err := feeLimit.Validate(); err != nil {
```

Comment:
I think this line is unnecessary because MsgRequestData already validated basic.

Reply:
Make sense, removed